cloud/infoblox

```diff
@@ -0,0 +1,278 @@
+				raise InfobloxNoIPavailableException(r_json['text'])
+			    else:
+				raise InfobloxGeneralException(r_json['text'])
+			else:
+			    r.raise_for_status()
+		else:
+		    raise InfobloxNotFoundException("No requested network found: " + network)
+	    else:
+		if 'text' in r_json:
+		    raise InfobloxGeneralException(r_json['text'])
+		else:
+		    r.raise_for_status()
+	except ValueError:
+	    raise Exception(r)
+	except Exception:
+	    raise    
+    def get_host_by_search(self, fqdn):
+	""" Implements IBA REST API call to retrieve host records by fqdn regexp filter
+	Returns array of host names in FQDN matched to given regexp filter
+	:param fqdn: hostname in FQDN or FQDN regexp filter
+	"""
+	rest_url = 'https://' + self.iba_host + '/wapi/v' + self.iba_wapi_version + '/record:host?name~=' + fqdn + '&view=' + self.iba_dns_view
+	hosts = []
+	try:
+            #print rest_url
+            #print self.iba_user
+            #print self.iba_verify_ssl
+           
+	    #r = requests.get(url=rest_url, auth=(self.iba_user, self.iba_password), verify=self.iba_verify_ssl)
+	    r = requests.get(url=rest_url, auth=(self.iba_user, self.iba_password), verify=False)
+	    r_json = r.json()
+	    if r.status_code == 200:
+		if len(r_json) > 0:
+		    for host in r_json:
+			hosts.append(host['name'])
+		    return hosts
+		else:
+		    raise InfobloxNotFoundException("No hosts found for regexp filter: " + fqdn)
+	    else:
+		if 'text' in r_json:
+		    raise InfobloxGeneralException(r_json['text'])
+		else:
+		    r.raise_for_status()
+	except ValueError:
+	    raise Exception(r)
+	except Exception:
+	    raise
+    def get_next_available_ip(self,network):
+        """ Implements IBA next_available_ip REST API call
+        Returns IP v4 address
+        :param network: network in CIDR format
+        """
+        rest_url = 'https://' + self.iba_host + '/wapi/v' + self.iba_wapi_version + '/network?network=' + network + '&network_view=' + self.iba_network_view
+        try:
+            r = requests.get(url=rest_url, auth=(self.iba_user, self.iba_password), verify=self.iba_verify_ssl)
+            r_json = r.json()
+            if r.status_code == 200:
+                if len(r_json) > 0:
+                    net_ref = r_json[0]['_ref']
+                    #Changed the num 1 to 10 for gettting 10 free ips 
+                    rest_url = 'https://' + self.iba_host + '/wapi/v' + self.iba_wapi_version + '/' + net_ref + '?_function=next_available_ip&num=5'
+                    r = requests.post(url=rest_url, auth=(self.iba_user, self.iba_password), verify=self.iba_verify_ssl)
+                    r_json = r.json()
+                    if r.status_code == 200:
+                        ip_v4 = r_json['ips']
+                        for i in range(len(ip_v4)): 
+                           ip_v4 = r_json['ips'][i]
+                           response = os.system("ping -c 1 -w2 " + ip_v4 + " > /dev/null 2>&1")
```

Comment:
This is not portable ( ping on freebsd do not support -w, so I assume that it doesn't work on OS X as well, and I wouldn't be surprised that windows is causing trouble )